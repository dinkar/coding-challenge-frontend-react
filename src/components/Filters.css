.filters__list {
    left: 0;
    top: 0;
    width: 100%;
    height: auto;
    padding: 0 2rem;
    display: flex;
    align-items: center;
    justify-content: center;
    box-sizing: border-box;
  }

  .filters__item {
    padding: 0.5rem;
    margin: 0.5rem;
  }

  