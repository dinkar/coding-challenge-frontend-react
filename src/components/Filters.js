import React from 'react';
import './Filters.css';

const filters = ({
  onUpdateSearchParams,
  occurred_after,
  occurred_before,
  query
}) => {
  const searchFromElRef = React.createRef();
  const searchToElRef = React.createRef();
  const searchDescriptionElRef = React.createRef();

  const handleSubmit = e => {
    e.preventDefault();
    const occurred_after = searchFromElRef.current.value
      ? +new Date(searchFromElRef.current.value) / 1000
      : '';
    const occurred_before = searchToElRef.current.value
      ? +new Date(searchToElRef.current.value) / 1000
      : '';
    const query = searchDescriptionElRef.current.value
      ? searchDescriptionElRef.current.value
      : '';
    onUpdateSearchParams({
      occurred_after,
      occurred_before,
      query
    });
  };

  return (
    <div>
      <form onSubmit={handleSubmit} className="filters__list">
        <div className="filters__item font-light">
          <label htmlFor="search-description">Search case description</label>
          <input
            type="text"
            placeholder="Search case description"
            ref={searchDescriptionElRef}
            id="search-description"
            defaultValue={query}
          />
        </div>
        <div className="filters__item font-light">
          <label htmlFor="search-from">From</label>
          <input
            type="datetime-local"
            placeholder="From"
            ref={searchFromElRef}
            id="search-from"
            defaultValue={
              occurred_after
                ? new Date(occurred_after * 1000).toISOString().split('.')[0]
                : ''
            }
          />
        </div>
        <div className="filters__item font-light">
          <label htmlFor="search-to">To</label>
          <input
            type="datetime-local"
            placeholder="To"
            ref={searchToElRef}
            id="search-to"
            max={new Date(Date.now()).toISOString().split('.')[0]}
            defaultValue={
              occurred_before
                ? new Date(occurred_before * 1000).toISOString().split('.')[0]
                : ''
            }
          />
        </div>
        <div className="filters__item">
          <input type="submit" value="Find Cases" className="btn" />
        </div>
      </form>
    </div>
  );
};

export default filters;
