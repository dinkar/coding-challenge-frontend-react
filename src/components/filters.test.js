import Filters from './Filters';
import renderer from 'react-test-renderer';
import React from 'react';

describe('<Fillters> tests', () => {
  let props;
  beforeEach(() => {
    props = {
      onUpdateSearchParams: jest.fn(),
      occurred_after: 1540400215, //Wed Oct 24 2018 22:26:55 GMT+0530 (India Standard Time)
      occurred_before: 1550501335, //Mon Feb 18 2019 20:18:55 GMT+0530 (India Standard Time)
      query: ''
    };
    Date.now = jest.fn(() => 1550501335275);
  });
  it('should match with the snapshot', () => {
    const component = renderer.create(<Filters props={props} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    tree.props.query = 'hello world';
    expect(tree).toMatchSnapshot();
  });
});
