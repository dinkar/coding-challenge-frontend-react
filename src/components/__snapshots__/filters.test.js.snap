// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Fillters> tests should match with the snapshot 1`] = `
<div>
  <form
    className="filters__list"
    onSubmit={[Function]}
  >
    <div
      className="filters__item font-light"
    >
      <label
        htmlFor="search-description"
      >
        Search case description
      </label>
      <input
        id="search-description"
        placeholder="Search case description"
        type="text"
      />
    </div>
    <div
      className="filters__item font-light"
    >
      <label
        htmlFor="search-from"
      >
        From
      </label>
      <input
        defaultValue=""
        id="search-from"
        placeholder="From"
        type="datetime-local"
      />
    </div>
    <div
      className="filters__item font-light"
    >
      <label
        htmlFor="search-to"
      >
        To
      </label>
      <input
        defaultValue=""
        id="search-to"
        max="2019-02-18T14:48:55"
        placeholder="To"
        type="datetime-local"
      />
    </div>
    <div
      className="filters__item"
    >
      <input
        className="btn"
        type="submit"
        value="Find Cases"
      />
    </div>
  </form>
</div>
`;

exports[`<Fillters> tests should match with the snapshot 2`] = `
<div
  query="hello world"
>
  <form
    className="filters__list"
    onSubmit={[Function]}
  >
    <div
      className="filters__item font-light"
    >
      <label
        htmlFor="search-description"
      >
        Search case description
      </label>
      <input
        id="search-description"
        placeholder="Search case description"
        type="text"
      />
    </div>
    <div
      className="filters__item font-light"
    >
      <label
        htmlFor="search-from"
      >
        From
      </label>
      <input
        defaultValue=""
        id="search-from"
        placeholder="From"
        type="datetime-local"
      />
    </div>
    <div
      className="filters__item font-light"
    >
      <label
        htmlFor="search-to"
      >
        To
      </label>
      <input
        defaultValue=""
        id="search-to"
        max="2019-02-18T14:48:55"
        placeholder="To"
        type="datetime-local"
      />
    </div>
    <div
      className="filters__item"
    >
      <input
        className="btn"
        type="submit"
        value="Find Cases"
      />
    </div>
  </form>
</div>
`;
